### DISCUSSION TITLE
Question about the getNeighborADMaterialProperty method

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#25081] [Q&A Modules: General]
Hi all, I want to know I can get the material property of which element if I use the getNeighborADMaterialProperty method?
1D mesh for example? 

Ethan

**Comment by GiudGiud** [comment]
Hello

getNeighborMaterialProperty is to be used in the context of side integrations on interfaces / within-domain boundaries. 

the neighbor element is on the other (exterior) side of the sideset, sidesets are oriented. 

Guillaume

**Reply by Ethan-xj** [reply]
Thanks Giud. In THM, `getNeighborMaterialProperty` is used in `ADNumericalFlux3EqnDGKernel` to calculate the flux between elements. But I'm confused.
As my understanding, for example, a 1D mesh which has 10 elements. Its id is from 0 to 9. If I use `getNeighborMaterialProperty` in elem0, I will get the material property of elem1, in elem1 and get mat prop of elem2, right? If so, what will I get if I use it in the last element?

**Reply by GiudGiud** [reply]
DGKernels also consider sides of elements.
So it s not on elem0, it s on elem0 and side "1" (for example)

**Reply by lindsayad** [reply]
@Ethan-xj DGKernels never operate on external boundaries

**Reply by Ethan-xj** [reply]
I got it. Thank you all :)
